src/mapgen/mapgen_v6.cpp

```diff
@@ -807,51 +806,44 @@ void MapgenV6::flowMud(s16 &mudflow_minpos, s16 &mudflow_maxpos)
 
 					VoxelArea::add_y(em, i, -1);
 				}
-
-				// Stop if out of area
-				//if(vmanip.m_area.contains(i) == false)
+				// No mud found in mapchunk column, process the next column
 				if (y < node_min.Y)
 					break;
 
 				if (n->getContent() == c_dirt ||
 						n->getContent() == c_dirt_with_grass) {
-					// Make it exactly mud
+					// Convert dirt_with_grass to dirt
 					n->setContent(c_dirt);
-
-					// Don't flow it if the stuff under it is not mud
+					// Don't flow it if the stuff under it is not mud,
+					// to leave at least 1 node of mud.
 					{
 						u32 i2 = i;
 						VoxelArea::add_y(em, i2, -1);
-						// Cancel if out of area
-						if (!vm->m_area.contains(i2))
-							continue;
 						MapNode *n2 = &vm->m_data[i2];
 						if (n2->getContent() != c_dirt &&
 								n2->getContent() != c_dirt_with_grass)
+							// Find next mud node in column
 							continue;
 					}
 				}
 
 				static const v3s16 dirs4[4] = {
-					v3s16(0, 0, 1), // back
-					v3s16(1, 0, 0), // right
-					v3s16(0, 0, -1), // front
-					v3s16(-1, 0, 0), // left
+					v3s16(0, 0, 1), // Back
+					v3s16(1, 0, 0), // Right
+					v3s16(0, 0, -1), // Front
+					v3s16(-1, 0, 0), // Left
 				};
 
-				// Check that upper is walkable. Cancel
-				// dropping if upper keeps it in place.
+				// Check if node above is walkable. If so, cancel
+				// flowing as if node above keeps it in place.
 				u32 i3 = i;
 				VoxelArea::add_y(em, i3, 1);
-				MapNode *n3 = NULL;
-
-				if (vm->m_area.contains(i3)) {
```

Comment:
Unnecessary check removed because:
At this point in code, y is no higher than node_max.Y (mapchunk top), voxelmanip top is 16 nodes above node_max.Y, so y + 1 is obviously inside area.